semantic/analyze.go

```diff
@@ -295,23 +297,33 @@ func analyzeMemberExpression(member *ast.MemberExpression) (*MemberExpression, e
 	if err != nil {
 		return nil, err
 	}
-
-	var propertyName string
-	switch p := member.Property.(type) {
-	case *ast.Identifier:
-		propertyName = p.Name
-	case *ast.StringLiteral:
-		propertyName = p.Value
-	case *ast.IntegerLiteral:
-		propertyName = strconv.FormatInt(p.Value, 10)
-	default:
-		return nil, fmt.Errorf("unsupported member property expression of type %T", member.Property)
-	}
-
 	return &MemberExpression{
 		loc:      loc(member.Location()),
 		Object:   obj,
-		Property: propertyName,
+		Property: member.Property.Name,
+	}, nil
+}
+
+func analyzeIndexExpression(e *ast.IndexExpression) (Expression, error) {
+	object, err := analyzeExpression(e.Object)
+	if err != nil {
+		return nil, err
+	}
+	index, err := analyzeExpression(e.Index)
+	if err != nil {
+		return nil, err
+	}
+	if n, ok := index.(*StringLiteral); ok {
+		return &MemberExpression{
```

Comment:
Ah I see how it is working now... I still think we should make the distinction at the parser level.

Reply:
Ah ok. Oops, disregard my comment above.